StepperEngine/src/Steps/Step.java

```diff
@@ -13,9 +13,9 @@ public abstract class  Step {
     private String alias;
     private Boolean hasAlias;
     private Boolean isReadOnly; // a readonly step doesn't change anything in the system.
-    private static double durationAvgInMs = 0.0;
-    private static double runTimeInMs = 0;
-    private static int stepRunsCounter = 0;
+    protected   double durationAvgInMs = 0.0;
+    protected   double runTimeInMs = 0;
+    protected   int startUpCounter = 0;
```

Comment:
why do we need those fields? can't we manage with just the static fields?

Reply:
i have searched on the web of parent class having static fiields, and his children using those fields and found this:

https://stackoverflow.com/questions/17172033/java-and-inherited-static-members

tldr; if for example we run the FileDumperStep once and run FilesReanamerStep once, we would think that both start up counter would be 1, but in actuality it will be 2.

thus, we can't store the static members in Step, but in his children